lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -1661,6 +1661,20 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
       return b.create<arith::MulIOp>(loc, lhs, rhs);
     }
   }
+  if (auto gtTensor = dyn_cast<AtenGtTensorOp>(op)) {
+    AtenGtTensorOp::Adaptor adaptor(operands);
+    Type dtype = payloadArgs[0].getType();
+
+    // TODO: Currently, the dominating dtype is the lhs type whereas the
+    // bitwidth of lhs and rhs should be compared to get the resulting dtype.
+    Value lhs = convertScalarToDtype(b, loc, payloadArgs[0], dtype);
+    Value rhs = convertScalarToDtype(b, loc, payloadArgs[1], dtype);
+    if (dtype.isa<mlir::FloatType>()) {
+      return b.create<arith::CmpFOp>(loc, arith::CmpFPredicate::UGT, lhs, rhs);
+    } else {
+      return b.create<arith::CmpIOp>(loc, arith::CmpIPredicate::sgt, lhs, rhs);
```

Comment:
I don't think this is exhaustive. In particular, `torch.uint8` is not handled correctly by this line because you're using the signed comparison. I would add an `else-if` branch checking if `dtype.isSignedInteger()` (https://mlir.llvm.org/doxygen/classmlir_1_1Type.html#a50385bcb919f49adab0f8f2521dc479f), and the else can return an error message saying that the only supported types at the moment are floats and signed ints.

Reply:
I see, torch.int32 is seen as a signless integer in payload args. Currently, I am only supporting float types.

Reply:
I have tried supporting the integer types as well. Please review.